Packs/ThreatMiner/Integrations/ThreatMiner/ThreatMiner.py

```diff
@@ -0,0 +1,698 @@
+    uris_rawdata = response.get('results', [])
+    return uris_rawdata
+
+
+def get_ip_URI(threatminer_results_as_array, max_returned_array_size):
+    uri_counter = 0
+    URIs = []
+
+    for uri in threatminer_results_as_array['raw_ip_uris']:
+        if max_returned_array_size == -1 or uri_counter < max_returned_array_size:
+            URIs.append({
+                'Address': threatminer_results_as_array['raw_ip_uris'][uri_counter]['uri'],
+                'LastSeen': threatminer_results_as_array['raw_ip_uris'][uri_counter]['last_seen']
+            })
+            uri_counter += 1
+        else:
+            break
+
+    return URIs
+
+
+def get_ip_MD5_rawdata(ip_address):
+    threat_miner_domain_url_postfix = 'host.php?q={}&rt={}'.format(ip_address, 4)
+    response = http_request('GET', THREAT_MINER_URL + threat_miner_domain_url_postfix, DEFAULT_HEADERS)
+
+    md5s = response.get('results', [])
+
+    if len(md5s) == 0:
+        return []
+    return md5s
+
+
+def get_ip_SSL_rawdata(ip_address):
+    threat_miner_domain_url_postfix = 'host.php?q={}&rt={}'.format(ip_address, 5)
+    response = http_request('GET', THREAT_MINER_URL + threat_miner_domain_url_postfix, DEFAULT_HEADERS)
+
+    ssls_raw_data = response.get('results', [])
+    return ssls_raw_data
+
+
+def create_ip_command_markdown(ip_address, context):
+    md = '## Threat_miner IP report for: {}\n'.format(ip_address)
+    threat_miner_found_results = False
+
+    if len(context['Whois']) != 0:
+        md += tableToMarkdown("Whois for {}".format(ip_address), context['Whois'],
+                              ['Address', 'Country', 'Org', 'Bgp', 'Reverse', 'ASN'])
+        threat_miner_found_results = True
+    if len(context['PassiveDNS']) != 0:
+        md += tableToMarkdown("PassiveDNS for {}".format(ip_address), context['PassiveDNS'],
+                              ['Domain', 'FirstSeen', 'LastSeen'])
+        threat_miner_found_results = True
+    if len(context['URI']) != 0:
+        md += tableToMarkdown("{} URIs".format(ip_address), context['URI'],
+                              ['Address', 'LastSeen'])
+        threat_miner_found_results = True
+    if len(context['MD5']) != 0:
+        md += tableToMarkdown("{} MD5s".format(ip_address), context['MD5'], ['MD5'])
+        threat_miner_found_results = True
+    if len(context['SSL']) != 0:
+        md += tableToMarkdown("{} SSLs".format(ip_address), context['SSL'], ['SSL'])
+        threat_miner_found_results = True
+
+    if not threat_miner_found_results:
+        md += 'No results found'
+
+    return md
+
+
+def get_ip_report_from_threat_miner(ip_address):
+    return {
+        'raw_whois': get_ip_whois_rawdata(ip_address),
+        'raw_passive_dns': get_ip_passiveDNS_rawdata(ip_address),
+        'raw_ip_md5': get_ip_MD5_rawdata(ip_address),
+        'raw_ip_uris': get_ip_URI_rawdata(ip_address),
+        'raw_ip_ssl': get_ip_SSL_rawdata(ip_address)
```

Comment:
The problem you fixed, returning an empty dictionary when there are no results returned from the request made instead of a list, does that problem also exist in all these other functions and not just `get_ip_whois_rawdata`? If so, shouldn't we fix it in all those other functions as well.

Reply:
And what about the `file` and `domain` commands... Need to check if the same problem is present in those commands and if so fix it.

Reply:
actually the rest can be lists, they are handled differently than WHOIS results

for some reason `file` and `domain` are handled, they do return `{}` in case no results were found - the issue was only with the `ip` command